osf/models/spam.py

```diff
@@ -221,6 +224,8 @@ def do_check_spam(self, author, author_email, content, request_headers, update=T
         if self.is_spammy:
             return True
 
+        enqueue_task(check_resource_for_domains.s(guid=self.guids.first()._id))
```

Comment:
> If we only extract the domains here, that means it will never extract domains for spammy or hammy resources. 

Yes this is expected behavior, once we've backfilled everything there's no need to recheck Spammed or Hammed resources.

> Also, are we sure that all models that inherits SpamMixin will have a guid?

Yes

Reply:
We have no guarantee that classes inheriting from Spam/SpamOverrideMixin also inherit from GuidMixin, but for now, they all do